scalafix-core/src/main/scala/scalafix/lint/LintMessage.scala

```diff
@@ -3,34 +3,88 @@ package scalafix.lint
 import scala.meta.Position
 import scalafix.rule.RuleName
 
-/** An observation of a LintCategory at a particular position
+/** A linter message reporting a code style violation.
   *
-  * @param message The message to display to the user. If empty, LintID.explanation
-  *                is used instead.
-  * @param position Optionally place a caret under a location in a source file.
-  *                 For an empty position use Position.None.
-  * @param category the LintCategory associated with this message.
+  * It's idiomatic to implement a custom class that extends this trait for each
+  * unique category of linting messages. For example, if you have an "unused code"
+  * linter then you might want to create a <code>class UnusedCode extends LintMessage</code>
+  * class with the appropriate context.
+  *
+  * Expensive values such as the message and explanation can be computed on-demand.
+  *
+  * @note for a LintMessage that is associated with a specific rule use
+  *       [[scalafix.lint.LintDiagnostic]].
   */
-final case class LintMessage(
-    message: String,
-    position: Position,
-    category: LintCategory
-) {
-
-  def format(explain: Boolean): String = {
-    val explanation =
-      if (explain)
-        s"""
-           |Explanation:
-           |${category.explanation}
-           |""".stripMargin
-      else ""
-
-    s"[${category.id}] $message$explanation"
+trait LintMessage {
```

Comment:
`LintMessage` became a trait so that it's possible (but not necessary) to implement custom message types

```scala
class UnusedCode(val position: Position) extends LintMessage { ... }
```

Reply:
Question whether basic values should have default implementations 🤔 For example `categoryID` and `explanation` could be empty string and `severity` could default to error. I'll go ahead and update that ASAP.

Reply:
Done.